platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConverterService.java

```
@@ -137,32 +137,82 @@ void clear() {
     public void init() {
         throwIfInitialized();
         // Primitives
-        addConverter(Integer.TYPE, INTEGER_CONVERTER);
-        addConverter(Long.TYPE, LONG_CONVERTER);
-        addConverter(Double.TYPE, DOUBLE_CONVERTER);
-        addConverter(Float.TYPE, FLOAT_CONVERTER);
-        addConverter(Short.TYPE, SHORT_CONVERTER);
-        addConverter(Byte.TYPE, BYTE_CONVERTER);
-        addConverter(Boolean.TYPE, BOOLEAN_CONVERTER);
-
-        addConverter(String.class, STRING_CONVERTER);
-        addConverter(Integer.class, INTEGER_CONVERTER);
-        addConverter(Long.class, LONG_CONVERTER);
-        addConverter(Double.class, DOUBLE_CONVERTER);
-        addConverter(Float.class, FLOAT_CONVERTER);
-        addConverter(Short.class, SHORT_CONVERTER);
-        addConverter(Byte.class, BYTE_CONVERTER);
-        addConverter(Boolean.class, BOOLEAN_CONVERTER);
-        addConverter(BigDecimal.class, BIG_DECIMAL_CONVERTER);
-        addConverter(BigInteger.class, BIG_INTEGER_CONVERTER);
-        addConverter(URL.class, URL_CONVERTER);
-        addConverter(URI.class, URI_CONVERTER);
-        addConverter(Path.class, PATH_CONVERTER);
-        addConverter(File.class, FILE_CONVERTER);
-        addConverter(ZonedDateTime.class, ZONED_DATE_TIME_CONVERTER);
-        addConverter(Duration.class, DURATION_CONVERTER);
-        addConverter(ChronoUnit.class, CHRONO_UNIT_CONVERTER);
-        addConverter(InetAddress.class, INET_ADDRESS_CONFIG_CONVERTER);
+        if(!converters.containsKey(Integer.TYPE)) {
```

Instead of having multiple `if` statements you can modify a `addConverter` like this: 

```
     if (converters.containsKey(converterType)) {
            if(initialized) { 
                 throw new IllegalStateException("Converter for type '" + converterType + "' already registered");
            } else {
               return; 
            }
            
        }
```

---

Good suggestion.
Note the impact here is only the first converter is cached and therefore utilized. Converters suggested in future calls will be silently ignored.

---

Updated